src/Pages/Signup/Signup.jsx

```diff
@@ -1,43 +1,127 @@
-import React from 'react';
+import { useState } from 'react';
 import { Link } from 'react-router-dom';
 import { Header } from '../../Components';
 import styles from '../Login/Login.module.css';
+import axios from 'axios';
+import { useAuth } from '../../Context';
+import { useNavigate } from 'react-router-dom';
 
 const Signup = () => {
+	const [formData, setFormData] = useState({
+		name: '',
+		email: '',
+		password: '',
+		confirmPassword: '',
+	});
+
+	const navigate = useNavigate();
+
+	const { authDispatch } = useAuth();
+
+	const formSubmitHandler = async (e) => {
+		e.preventDefault();
+
+		if (formData.name && formData.email && formData.password && formData.confirmPassword) {
+			if (formData.password === formData.confirmPassword) {
+				try {
+					const response = await axios.post('/api/auth/signup', {
+						email: formData.email,
+						password: formData.password,
+						name: formData.name,
+					});
+					if (response.status === 201) {
+						authDispatch({
+							type: 'SIGNUP',
+							payload: { token: response.data.encodedToken, user: response.data.createdUser },
+						});
+						localStorage.setItem('token', response.data.encodedToken);
+						localStorage.setItem('user', JSON.stringify(response.data.createdUser));
+						navigate('/');
+						alert('Signed up');
+					} else {
+						console.error('ERROR: ', response);
```

Comment:
You can remove this console.log

Reply:
These are intentional console.logs for showing error, if there is any, to the reviewer. The one's mentioned below are also intentional.